pilot/pkg/bootstrap/validation.go

```diff
@@ -100,16 +75,23 @@ func (s *Server) initConfigValidation(args *PilotArgs) error {
 			WebhookConfigName: webhookConfigName,
 			ServiceName:       "istiod",
 		}
-		whController, err := controller.New(o, s.kubeClient, dynamicInterface)
-		if err != nil {
-			return err
-		}
 		s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
 			leaderelection.
-				NewLeaderElection(args.Namespace, args.PodName, leaderelection.ValidationController, s.kubeClient).
-				AddRunFunction(func(stop <-chan struct{}) {
+				NewLeaderElection(args.Namespace, args.PodName, leaderelection.ValidationController, s.kubeClients).
+				AddRunFunction(func(leaderStop <-chan struct{}) {
+					whController, err := controller.New(o, s.kubeClients)
+					if err != nil {
+						log.Errorf("failed to start validation controller")
+						return
+					}
 					log.Infof("Starting validation controller")
-					whController.Start(stop)
+					// Start informers again. This fixes the case where informers for namespace do not start,
+					// as we create them only after acquiring the leader lock
+					// Note: stop here should be the overall pilot stop, NOT the leader election stop. We are
+					// basically lazy loading the informer, if we stop it when we lose the lock we will never
```

Comment:
Wouldn't no re-creating the informers lead to cases where a leader is elected who is not receiving updates?

Reply:
Every time the leader is elected, this function runs which will grab the existing informer or create a new one if its not present, then start all unstarted informers. So I think this should work? Can you give some more details of where it would break down?

Reply:
I misread your comment. This looks fine.